src/gpuarray_elemwise.c

```diff
@@ -72,6 +79,41 @@ static void free_args(unsigned int n, gpuelemwise_arg *args) {
   free(args);
 }
 
+#define MUL_NO_OVERFLOW ((size_t)1 << (sizeof(size_t) * 4))
+
+static int reallocaz(void **p, size_t e, size_t o, size_t n) {
+  char *res;
+
+  assert(o <= n);
+
+  if ((n >= MUL_NO_OVERFLOW || e >= MUL_NO_OVERFLOW) &&
+      n > 0 && SIZE_MAX / n < e) {
+    return 1;
+  }
+  res = realloc(*p, e*n);
+  if (res == NULL) return 1;
+  memset(res + (e*o), 0, e*(n-o));
+  *p = (void *)res;
+  return 0;
+}
+
+static int ge_grow(GpuElemwise *ge, unsigned int nd) {
+  unsigned int i;
+
+  assert(nd > ge->nd);
```

Comment:
Assert in C are for things that should always be true but if they are not are indicator of a more serious problem.

I don't want a pretty user error message in this case since the only reason that would be false is a bug.  Also, this way the check is only done in the debug build.